fiona/__init__.py

```diff
@@ -60,7 +62,7 @@
     _remove,
     _remove_layer,
 )
-from fiona.path import ParsedPath, parse_path, vsi_path
+from fiona._path import _ParsedPath, _UnparsedPath, _parse_path, _vsi_path
```

Comment:
FWIW, I see you added a deprecation warning inside the `fiona.path` module. But the above change directly breaks code that was using those functions from top-level `fiona` instead of from `fiona.path`

We are running into some issues with this in geopandas: https://github.com/geopandas/geopandas/issues/3207

Reply:
I commented at https://github.com/geopandas/geopandas/issues/3207. Fiona's path module and methods are only public by accident. Other projects shouldn't be coupled to fiona's internals, which are full of weird details. Removing this coupling between geopandas and fiona shouldn't be hard and I'm happy to help.